### PR TITLE
Upgrades unit tests to XCTest and ARC.

### PR DESCRIPTION
**Pull request by StevenEWright**
PR Upgrades CoreParseTests and iOSCoreParseTests to XCTest from SenTest. Also updates unit tests (only) to ARC.

### CODE REVIEW

File: CoreParseTests/CoreParseTests.m [hunk 1]
```diff
@@ -577,9 +589,7 @@ - (void)testMapCSSTokenisationPt
                                                               [CPKeywordToken tokenWithKeyword:@"}"],
                                                               [CPEOFToken eof],
                                                               nil]];
-    STAssertEqualObjects(t1, t2, @"Tokenised MapCSS with size specifier incorrectly", nil);
-    
-    [self tearDownMapCSS];
```
Comment (StevenEWright) [inline CoreParseTests/CoreParseTests.m]: Oops, I shouldn't have remove the tearDownMapCSS call. Let me fix that.

File: CoreParseTests/CoreParseTests.m [hunk 2]
```diff
@@ -587,31 +597,28 @@ - (void)testMapCSSParsing
     [self setUpMapCSS];
     CPSyntaxTree *tree = [mapCssParser parse:[mapCssTokeniser tokenise:mapCssInput]];
     
-    STAssertNotNil(tree, @"Failed to parse MapCSS", nil);
-    
-    [self tearDownMapCSS];
```
Comment (StevenEWright) [inline CoreParseTests/CoreParseTests.m]: Ditto.

File: CoreParseTests/CoreParseTests.m [hunk 3]
```diff
@@ -587,31 +597,28 @@ - (void)testMapCSSParsing
...
     [self setUpMapCSS];
-    CPTokenStream *stream = [[[CPTokenStream alloc] init] autorelease];
+    CPTokenStream *stream = [[CPTokenStream alloc] init];
     [NSThread detachNewThreadSelector:@selector(runMapCSSTokeniser:) toTarget:self withObject:stream];
     CPSyntaxTree *tree1 = [mapCssParser parse:stream];
     CPSyntaxTree *tree2 = [mapCssParser parse:[mapCssTokeniser tokenise:mapCssInput]];
     
-    STAssertEqualObjects(tree1, tree2, @"Parallel parse of MapCSS failed", nil);
-    
-    [self tearDownMapCSS];
```
Comment (StevenEWright) [inline CoreParseTests/CoreParseTests.m]: Again :/

File: CoreParse.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1319,89 +1237,165 @@
...
-					"-all_load",
-				);
-				PRODUCT_NAME = CoreParseTests;
-				SDKROOT = iphoneos;
-				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
-				WRAPPER_EXTENSION = octest;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/../Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.10;
+				MTL_ENABLE_DEBUG_INFO = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = com.stevenewright.CoreParseTests;
```
Comment (StevenEWright) [inline CoreParse.xcodeproj/project.pbxproj]: Didn't mean to add these com.stevenewright.* identifiers. I'll swap them out for what you had before. Here, and below.

File: CoreParse.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1319,89 +1237,165 @@
...
-				);
-				PRODUCT_NAME = CoreParseTests;
-				SDKROOT = iphoneos;
-				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator";
-				VALIDATE_PRODUCT = YES;
-				WRAPPER_EXTENSION = octest;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/../Frameworks @loader_path/../Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.10;
+				MTL_ENABLE_DEBUG_INFO = NO;
+				PRODUCT_BUNDLE_IDENTIFIER = com.stevenewright.CoreParseTests;
```
Comment (StevenEWright) [inline CoreParse.xcodeproj/project.pbxproj]: (Here)

File: CoreParse.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1319,89 +1237,165 @@
...
+					"DEBUG=1",
+					"$(inherited)",
+				);
+				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				INFOPLIST_FILE = "CoreParseTests/CoreParseTests-Info.plist";
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				MTL_ENABLE_DEBUG_INFO = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = com.stevenewright.iOSCoreParseTests;
```
Comment (StevenEWright) [inline CoreParse.xcodeproj/project.pbxproj]: (And here)

File: CoreParse.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -1319,89 +1237,165 @@
...
+				ENABLE_NS_ASSERTIONS = NO;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				INFOPLIST_FILE = "CoreParseTests/CoreParseTests-Info.plist";
+				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				MTL_ENABLE_DEBUG_INFO = NO;
+				PRODUCT_BUNDLE_IDENTIFIER = com.stevenewright.iOSCoreParseTests;
```
Comment (StevenEWright) [inline CoreParse.xcodeproj/project.pbxproj]: (Last one)

**Comment by beelsebob** [comment]
I'd be very happy to accept a patch to upgrade to XCTest.  However CoreParse is an MRC framework (with good reason, performance under ARC was significantly regressed).  I'd like to keep the project consistent in that regard (even though the tests can be converted separately).

**Comment by StevenEWright** [comment]
Hey Tom. Thanks for the reply. I can understand why you'd feel that way. I had seen your previous rejection of a pull request to upgrade the entire project to ARC so I wasn't sure how you'd take this one.

I'd (politely) challenge your desire for consistency by pointing out a few things;

- It is best if tests are as easy to create/update/maintain as possible. People will continue to keep them working (and therefore useful) IIF working on them is easy. Converting them to ARC helps keep them easy to work with.

- There are several memory leaks in the unit tests as they stand now. Specifically, most/all of the delegates get alloc'ed but never released (until the tests process dies.) If they are kept as non-ARC, then these correctness issues should probably be addressed. 

While making the unit tests different from the rest of the project may seem oddly inconsistent, I really believe it'd a useful pragmatic improvement (everyone's OCD aside.)

Let me know if you feel super strongly about the ARC thing in the context of the unit tests. If so, let's reject this PR for now (it was late last night when I sent it to you, and it has a few other problems which were bothering me; see all my comments and some inconsistent formatting that I'd need to fix anyways.)

Thanks again for taking a look! Let me know.

**Comment by StevenEWright** [comment]
FYI; Fixed the outstanding issues.

**Comment by beelsebob** [comment]
Yeh, you're right.  Correctness is always more important.  Thanks a lot for putting the work in.

**Pull request merged**